### ISSUE TITLE
Reduce the cost of opnorm of $$B_k$$

### ISSUE THREAD

**Issue by MohamedLaghdafHABIBOULLAH** [JuliaSmoothOptimizers/RegularizedOptimization.jl#133]
"It could be interesting to try other norms that are less cheap than the spectral norm, particularly the trace norm (the sum of eigenvalues when $$B_k>0$$)

**Comment by dpo** [comment]
The Frobenius norm is also interesting because $\Vert A \Vert_2 \leq \Vert A \Vert_F \leq \Vert A \Vert_*$, and it’s cheap to compute.

**Comment by dpo** [comment]
I think the convergence theory continues to hold if we compute any $\beta_k = \Theta(\Vert B_k \Vert)$, but we have to decide on how to compute $\beta_k$, especially for quasi-Newton operators.

**Comment by MohamedLaghdafHABIBOULLAH** [comment]
If we use the Frobenius norm, then we solve the allocation issue with opnrom.

**Comment by MohamedLaghdafHABIBOULLAH** [comment]
Ok, I thought that we have norm for LinearOperators of all type...

**Comment by MohamedLaghdafHABIBOULLAH** [comment]
Otherwise, the trace norm if B_k is positive, and I believe that we can easily extract the diagonal

**Comment by dpo** [comment]
For the trace norm, we need the singular values. It's even costlier.

**Comment by MohamedLaghdafHABIBOULLAH** [comment]
Nope, if B_k is symmetric positive, then the trace norm is just the sum of its diagonal which is a norm!

**Comment by MaxenceGollier** [comment]
We should try to close this now to remove the allocations from R2N, @dpo you suggested using a Rayleigh quotient, can you try to implement it @MohamedLaghdafHABIBOULLAH ?
